homework-submission/week1/src/components/Todos.js

```diff
@@ -0,0 +1,34 @@
+import React, { Component } from "react";
+import List from "./List";
+import Checkbox from "./Checkbox";
+import RemoveTodo from "./RemoveTodo";
+
+class Todos extends Component {
+  render() {
+    return (
+      <div>
+        <ul>
+          {this.props.items.map((item, id) => (
+            <div key={id} className={item.done? 'checked' : ''}>
+            <Checkbox 
+              id = {item.id} 
+              checkbox = {item.done} 
+              handler = {this.props.handler}
+             />
+
+            <List 
+              description = {item.description} 
+              deadline = {item.deadline}
+             />
+             <RemoveTodo
+             remove = {this.props.remove}
```

Comment:
There's a better way to bind your `this.props.remove` function with the **id** without passing it to the `<RemoveTodo>`

```
<RemoveTodo
  remove={this.props.remove.bind(null, i)}
>
```

This `this.props.remove.bind(null, i)` will return a new function with `id` already stored inside it.

The `.bind` take the `id` store this id (somehow) until the function get called then it send the id by its own.

You can read about `bind` and what happens under the hood.